### DISCUSSION TITLE
Set value of NGINX remote_user variable from plugin

### DISCUSSION THREAD

**Discussion by shawnchapla-8451** [Kong/kong#10184] [Help] [answered]
We have a bespoke Lua plugin which performs custom authentication, which has the authenticated user account at request processing time.  I'd like to be able to use that value to set the NGINX remote_user variable so that it will be captured in the standard Kong access log along with the rest of the other log fields when that is written.  I know I can use the log module to write my own log custom entries, but it would be cleaner to leverage the remote_user field if possible.

It looks like the [nginx.lua](https://github.com/Kong/kong/blob/master/kong/pdk/nginx.lua) module exposes information about NGINX that can be _read_, but not to set anything.  Is that even possible?

Thanks in advance.

**Comment by chronolaw** [comment, marked as answer]
Yes, many Nginx vars are read only, and only a few are writeable.  

But We can use `set` directive to define a var, then manipulate it with `ngx.var`.

**Reply by shawnchapla-8451** [reply]
Thanks, that worked great!
